app/javascript/redux/greetings/greetingsSlice.js

```diff
@@ -0,0 +1,42 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+// we get the initial state from the greetings Api
+export const getGreetings = createAsyncThunk('greetings/getGreetings', async () => {
+    try {
+        const response = await axios.get('http://localhost:3000/api/v1/greetings');
+        return response.data;
+    } catch (error) {
+        console.error(error);
+    }
+});
+
```

Comment:
### 👋 Hey there! 👋

- [ ] I experienced the CORS policy error, and I think I've found a simple fix for it. The error is happening because of the Same-Origin Policy in web browsers. When we use an absolute URL like `http://localhost:3000/api/v1/greetings`, it's considered a cross-origin request, which triggers the CORS policy.

- [ ] To resolve this, we can simply use `/api/v1/greetings` because we're working on the same app, and there's no need for the absolute URL. This change should make the request a same-origin one, eliminating the CORS policy error. It's a more elegant solution for our setup.

![image](https://github.com/Diegogagan2587/hello_rails_react/assets/98453979/e0c48172-0306-447f-95a1-76442d6b20bd)


#### Happy coding! 🚀👨‍💻

Reply:
@frank1738  Thanks for pointing that out! I'll work on that, thanks so much!